### PR TITLE
Also forward error requests to the proxy

### PR DESCRIPTION
**Pull request by OliverJAsh**
Fixes https://github.com/webpack/webpack-dev-server/issues/2380

- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

Yes

### Motivation / Use-Case

See https://github.com/webpack/webpack-dev-server/issues/2380 and https://github.com/webpack/webpack-dev-server/issues/2380#issuecomment-581378628.

### CODE REVIEW

File: lib/Server.js [hunk 1]
```diff
@@ -319,7 +319,11 @@ class Server {
         } else {
           next();
         }
-      });
+      };
+
+      this.app.use(handle);
+      // Also forward error requests to the proxy so it can handle them.
+      this.app.use((error, req, res, next) => handle(req, res, next));
```
Comment (evilebottnawi) [inline lib/Server.js]: It is create two instances of proxy that was reduce performance

Comment (OliverJAsh) [reply lib/Server.js]: I don't see how that will hurt performance.

The proxy middleware is created once, outside of `handle`:

```ts
      proxyMiddleware = getProxyMiddleware(proxyConfig);
```

… and the result is used in `handle` on each request, be it a non-error or error request.

Comment (evilebottnawi) [reply lib/Server.js]: Each request now run `handle` (https://github.com/webpack/webpack-dev-server/blob/29a822ea8426820617745b3f355c344e84f7c01b/lib/Server.js#L291)  twice so we potentially send require to middleware twice https://github.com/webpack/webpack-dev-server/blob/29a822ea8426820617745b3f355c344e84f7c01b/lib/Server.js#L318

Comment (OliverJAsh) [reply lib/Server.js]: I don't believe that's true. `handle` will only run once:

- If the request has not errored, `handle` will be called via `app.use(handle)`.
- If the request has errored, `handle` will be called via `app.use((error, req, res, next) => handle(req, res, next))`.

You can see this for yourself if you add some logging to `handle`.

Comment (evilebottnawi) [reply lib/Server.js]: Hm, looks you are right https://expressjs.com/en/guide/error-handling.html

**Comment by evilebottnawi** [comment]
Something broken on CI, I rerun tests

**Comment by evilebottnawi** [comment]
Bit thanks, sometimes I get too much error flow, and do not have time to investigate them in detail, I apologize for the situation and am very glad that you yourself could find the problem and fix it

**Comment by OliverJAsh** [comment]
@evilebottnawi No problem, thanks for reviewing my PR! 😄

**Pull request merged**